src/ui/server/controllers/insurance/business/broker/validation/rules/broker-postcode.test.ts

```diff
@@ -0,0 +1,49 @@
+import brokerPostcode from './broker-postcode';
+import { ERROR_MESSAGES } from '../../../../../../content-strings';
+import FIELD_IDS from '../../../../../../constants/field-ids/insurance/exporter-business';
+import { RequestBody } from '../../../../../../../types';
+import postCodeValidation from '../../../../../../shared-validation/postcode';
+
+const {
+  BROKER: { POSTCODE, USING_BROKER },
+} = FIELD_IDS;
+
+const { EXPORTER_BUSINESS } = ERROR_MESSAGES.INSURANCE;
+const ERROR_MESSAGE = EXPORTER_BUSINESS[POSTCODE];
+
+describe('controllers/insurance/business/broker/validation/rules/broker-postcode', () => {
+  const mockErrors = {
+    summary: [],
+    errorList: {},
+  };
+
+  const mockBody = {
+    [POSTCODE]: '',
+  } as RequestBody;
+
+  it('should return the result of emptyFieldValidation if using broker is "Yes"', () => {
+    mockBody[USING_BROKER] = 'Yes';
+
+    const response = brokerPostcode(mockBody, mockErrors);
+
+    const expected = postCodeValidation(POSTCODE, mockBody[POSTCODE], ERROR_MESSAGE.IS_EMPTY, ERROR_MESSAGE.INCORRECT_FORMAT, mockErrors);
+
+    expect(response).toEqual(expected);
+  });
+
+  it('should return the mockErrors if using broker is "No"', () => {
+    mockBody[USING_BROKER] = 'No';
+
+    const response = brokerPostcode(mockBody, mockErrors);
+
+    expect(response).toEqual(mockErrors);
+  });
+
+  it('should return the mockErrors if using broker is "null"', () => {
+    mockBody[USING_BROKER] = null;
+
+    const response = brokerPostcode(mockBody, mockErrors);
+
+    expect(response).toEqual(mockErrors);
+  });
```

Comment:
Can we wrap the yes/no/null parts into describe blocks please like `broker-email`, easier to read/digest 😊